adit/upload/static/upload/upload.js

```diff
@@ -0,0 +1,361 @@
+    },
+
+    getFiles: function () {
+      var inputElement = formEl.querySelector("#fileselector");
+
+      if (
+        inputElement instanceof HTMLInputElement &&
+        inputElement.files.length > 0
+      ) {
+        return inputElement.files;
+      } else if (this.droppedFiles.length > 0) {
+        const files = [];
+        for (const f of this.droppedFiles) {
+          files.push(f);
+        }
+        return files;
+      } else {
+        return [];
+      }
+    },
+
+    toggleUploadButtonVisibility: function () {
+      // Check if files are selected
+      const files = this.getFiles();
+      this.buttonVisible = files.length > 0 ? true : false;
+      this.fileCount = files.length;
+      this.uploadCompleteTextVisible = false;
+    },
+    clearFiles: function () {
+      var inputEl = formEl.querySelector("#fileselector");
+
+      if (inputEl instanceof HTMLInputElement) {
+        inputEl.value = null;
+      }
+      this.droppedFiles = [];
+      this.toggleUploadButtonVisibility();
+    },
+
+    fileHandler: async function (fileObj, datasets) {
+      const arrayBuffer = await fileObj.arrayBuffer(); //await fileReader.readAsArrayBuffer(fileObj);
+      datasets.push(arrayBuffer);
+    },
+
+    chooseFolder: function () {
+      const files = this.getFiles();
+      this.loadFiles(files);
+    },
+
+    loadFiles: async function (files) {
+      const destinationSelect = formEl.querySelector('[name="destination"]');
+
+      if (destinationSelect instanceof HTMLSelectElement) {
+        const selectedOption =
+          destinationSelect.options[destinationSelect.selectedIndex];
+        var node_id = selectedOption.getAttribute("data-node_id");
+      }
+
+      if (files.length === 0) {
+        showToast("warning", "Sandbox", `No files selected.${files}`);
+      } else {
+        var datasets = [];
+        for (const fileEntry of files) {
+          await this.fileHandler(fileEntry, datasets);
+        }
+
+        let status = 0;
+        let loadedFiles = 0;
+
+        try {
+          const checker = await this.checkPatientIDs(datasets);
+
+          if (checker) {
+            const anon = this.createAnonymizer();
+
+            this.buttonVisible = false;
+            this.stopUploadVar = false;
+
+            for (const set of datasets) {
+              // Anonymize data and write back to bufferstream
+              const dicomData = dcmjs.data.DicomMessage.readFile(set, {
+                ignoreErrors: true,
+              });
+
+              await anon.anonymize(dicomData);
+              const anonymized_set = await dicomData.write();
+
+              this.pbVisible = true;
+
+              this.stopUploadButtonVisible = true;
+              if (this.stopUploadVar) {
+                // Stop uploading if stop button is clicked
+                break;
+              }
+
+              // Upload data to server
+              status = await uploadData({
```

Comment:
hier wird ein Object und kein python dict zurückgegeben, die keys brauchen in JS weder Klammern, noch Hochkomma